Update docs with info/examples on setting Out-File width via PSDefaultParameterValues (#7656)
MicrosoftDocs/PowerShell-Docs: reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md

formatted correctly. To write to files with a different encoding, use the
`Out-File` cmdlet with its `Encoding` parameter.

### Width of output when writing to a file

When you are writing to a file using either `Out-File` or the redirection
operators, PowerShell formats table output to the file based on the width of
the console it is running within. For instance, when logging table output

> to file using a command like e.g. `Get-ChildItem Env:\Path > path.log` on a
> system where the console width is set to 80, the output in the file is
> truncated to 80 characters:

Reviewer: ```suggestion
to file using a command like `Get-ChildItem Env:\Path > path.log` on a system 
where the console width is set to 80 columns, the output in the file is
truncated to 80 characters:
```

Author: Done

> ```text

Reviewer: ```suggestion
```Output
```

Name                         Value
----                         -----
Path                         C:\Program Files\PowerShell\7-preview;C:\WINDOWS…
```

Considering that the console width may be set arbitrarily on systems where
your script is run, you may prefer that PowerShell format table output to
files based on a width that you specify instead.

> The `Out-File` cmdlet provides a `Width` parameter that allows you to set the

Reviewer: ```suggestion
The `Out-File` cmdlet provides a **Width** parameter that allows you to set the
```

> width you would like for table output. Rather than having to add

Reviewer: The description is very focused on table output. It makes it sound like the width only affects tables, but the width applies to any output.

Reword this to make it clearer that the width applies to all output and that you are using Format-Table as an example.

Author: My experiments indicate it doesn't impact Format-List output e.g.:
```powershell
$obj = [pscustomobject]@{prop1=("*" * 1000); prop2=("-" * 1000)}
$obj | format-list > obj.log
notepad .\obj.log
```
You'll see in notepad that each line is 1009 in length without doing anything to the default width. I see the same thing if I use `Format-Custom` instead of `Format-List`.   And if I send a string to output like this:
```powershell
$obj | format-list > obj.log
notepad .\obj.log
```
I see a line in Notepad that ends at col 1001.  I'll bring this up in the wg-engine discord channel to verify.

`-Width 2000` everywhere you invoke `Out-File`, you can use the
`$PSDefaultParameterValues` variable to set this value for all usages of the
`Out-File` cmdlet in a script. And since the redirection operators (`>` and
`>>`) are effectively aliases for `Out-File`, setting the `Out-File:Width`

> parameter for the whole script impacts the formatting width for the

Reviewer: Should we make it clear that adding `$PSDefaultParameterValues` in a script makes the change local to the script, but running it from the command line is a global change?

Author: I think you have the common case of "user copies script from help example" covered below.  I'll turn that example into a function as you suggest.  Do you think that will be sufficient?

Reviewer: Add a link to the about topic.

```
For more information about `$PSDefaultParameterValues`, see [about_Preference_Variables](/powershell/module/microsoft.powershell.core/about/about_preference_variables#psdefaultparametervalues)
```

Author: I think the link can be simplified to `about_preference_variables.md#psdefaultparametervalues` based on what I see for the next link in this topic:
```
The `>` operator is not to be confused with the
[Greater-than](about_Comparison_Operators.md#-gt--ge--lt-and--le) comparison
```
If that's not the case let me know and I'll change it.

Reviewer: Yep, you are correct. That will work.

`-Width 2000` everywhere you invoke `Out-File`, you can use the
`$PSDefaultParameterValues` variable to set this value for all usages of the
`Out-File` cmdlet in a script. And since the redirection operators (`>` and
`>>`) are effectively aliases for `Out-File`, setting the `Out-File:Width`
parameter for the whole script impacts the formatting width for the
redirection operators as well. Put the following command near the top of your
script to set `Out-File:Width` for the whole script:

```powershell
$PSDefaultParameterValues['out-file:width'] = 2000
```

Increasing the output width will increase memory consumption when logging
table formatted output. If you are logging a lot of tabular data to file and
you know you can get by with a smaller width, use the smaller width.

In some cases, such as `Get-Service` output, in order to use the extra width
you will need to pipe the output through `Format-Table -AutoSize` before
outputting to file.

```powershell
$PSDefaultParameterValues['out-file:width'] = 2000
Get-Service | Format-Table -AutoSize > services.log
```

For more information about `$PSDefaultParameterValues`, see

> [about_Preference_Variables](about_preference_variables.md#psdefaultparametervalues)

Reviewer: ```suggestion
[about_Preference_Variables](about_preference_variables.md#psdefaultparametervalues).
```
